python/lsst/meas/extensions/scarlet/scarletDeblendTask.py

```diff
@@ -592,10 +596,16 @@ def deblend(self, mExposure, sources):
                 self._skipParent(src, mask)
                 self.log.trace('Parent %i: skipping masked footprint', int(src.getId()))
                 continue
-            if len(peaks) > self.config.maxNumberOfPeaks:
+            if self.config.maxNumberOfPeaks > 0 and len(peaks) > self.config.maxNumberOfPeaks:
```

Comment:
It doesn't particularly matter much but you could store this value before the loop rather than checking every time, or simply set maxNumberOfPeaks to np.inf if it's zero.